File: python/src/data_trainer.py
from format import __to_json__
import os
import datetime
import sys
sys.path.append('./src/lib')


class DataTrainer:
    def __init__(self, data_trainer: str) -> None:
        self.data_trainer = data_trainer

    def export(self, dirname: str) -> None:
        with open(self.__filename__(dirname), 'w') as f:
            f.write(__to_json__(self.data_trainer))

    # private

    def __filename__(self, dirname: str) -> str:
        return os.path.join(
            dirname, f'training_data_{
                datetime.datetime.now():%Y%m%d%H%M%S}.json')

File: ruby/src/data_trainer.rb
# rbs_inline: enabled

require_relative './lib/format'

class DataTrainer
  include ::Lib::Format

  # @rbs data_trainer: String
  # @rbs return: void
  def initialize(data_trainer)
    @data_trainer = data_trainer
  end

  # @rbs dirname: String
  # @rbs return: void
  def export(dirname)
    File.write(filename(dirname), to_json(data_trainer))
  end

  private

  attr_reader :data_trainer

  # @rbs dirname: String
  # @rbs return: String
  def filename(dirname)
    File.join(dirname, "training_data_#{Time.now.strftime('%F%T').gsub(/[:\-]/, '')}.json")
  end
end
